internal/infrastructure/middleware/enforce_identity_test.go

```diff
@@ -240,33 +241,52 @@ func TestEnforceIdentitySkipper(t *testing.T) {
 		err  error
 	)
 
+	userXRHID := api.NewUserXRHID().
+		WithOrgID("12345").
+		WithUsername("test-fail-predicate").
+		Build()
+
 	// When skipper return false, as no x-rh-identity provided, will return unauthorized
 	e = helperNewEchoEnforceIdentity(
 		EnforceIdentityWithConfig(
 			&IdentityConfig{
 				Skipper: helperSkipper(false),
+				Predicates: []IdentityPredicateEntry{
+					{
+						Name:      "test-fail-predicate",
+						Predicate: helperCreatePredicate("test-fail-predicate"),
+					},
+				},
 			},
 		),
 	)
 	res = httptest.NewRecorder()
 	req = httptest.NewRequest(http.MethodGet, "/test", nil)
+	req.Header.Add(header.HeaderXRHID, header.EncodeXRHID(&userXRHID))
```

Comment:
This line does not match the `e` comment: `// When skipper return false, as no x-rh-identity provided, will return unauthorized` I.e. the identity is provided. 

Btw the test seems to still pass when the modifications on lines 254-259 and 265 are removed.

Reply:
> This line does not match the `e` comment: `// When skipper return false, as no x-rh-identity provided, will return unauthorized` I.e. the identity is provided.

Good catch.  I updated the comments.

> Btw the test seems to still pass when the modifications on lines 254-259 and 265 are removed.

Yes, but it will pass for the wrong reason :)  Due to the changes, this would pass because the `ParseXRHID` middleware returns 401 because it does not see the `X-RH-Identity` header.  We want it to pass because the `EnforceIdentity` predicate fails (because the skipper *doesn't* skip the middleware).